metis-common/src/main/java/eu/europeana/metis/mongo/RecordDao.java

```diff
@@ -0,0 +1,161 @@
+package eu.europeana.metis.mongo;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.client.MongoClient;
+import dev.morphia.Datastore;
+import dev.morphia.Morphia;
+import dev.morphia.mapping.Mapper;
+import dev.morphia.mapping.MappingException;
+import dev.morphia.query.experimental.filters.Filters;
+import eu.europeana.corelib.definitions.edm.beans.FullBean;
+import eu.europeana.corelib.edm.exceptions.MongoDBException;
+import eu.europeana.corelib.edm.exceptions.MongoRuntimeException;
+import eu.europeana.corelib.edm.model.metainfo.WebResourceMetaInfoImpl;
+import eu.europeana.corelib.solr.bean.impl.FullBeanImpl;
+import eu.europeana.corelib.solr.entity.AgentImpl;
+import eu.europeana.corelib.solr.entity.AggregationImpl;
+import eu.europeana.corelib.solr.entity.BasicProxyImpl;
+import eu.europeana.corelib.solr.entity.ConceptImpl;
+import eu.europeana.corelib.solr.entity.ConceptSchemeImpl;
+import eu.europeana.corelib.solr.entity.EuropeanaAggregationImpl;
+import eu.europeana.corelib.solr.entity.EventImpl;
+import eu.europeana.corelib.solr.entity.PhysicalThingImpl;
+import eu.europeana.corelib.solr.entity.PlaceImpl;
+import eu.europeana.corelib.solr.entity.ProvidedCHOImpl;
+import eu.europeana.corelib.solr.entity.ProxyImpl;
+import eu.europeana.corelib.solr.entity.TimespanImpl;
+import eu.europeana.corelib.solr.entity.WebResourceImpl;
+import eu.europeana.corelib.web.exception.EuropeanaException;
+import eu.europeana.corelib.web.exception.ProblemType;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Connection for accessing Europeana records.
+ */
+public class RecordDao {
```

Comment:
The api is already depending on metis-common
I think this might be something we can think in the future.
I feel that  having a separate project just for this class is also too much, don't you think?

But in general I've thought of this as a whole because there are a few possibilities we can consider(no need to do this now imo):

- Either we create a separate project with libs that are used solely outside of metis and we have a metis-common to be "common" only for metis modules, but I think this will cause more issues similar to corelib.

- Another approach is to have more granular division of packages and:
   - Have instructions on the readme file with what to "exclude" if only part of the metis-common is to be used
   - Or make all dependencies "optional" and again in the readme specify what dependencies to "declare" to use the 
     desired functionality of from metis-common.

What do you think?

(Generally the idea is to move stuff around for now and minimize changes on the class level)

Reply:
First, I understand that you're just moving things in, but I do also believe that there is some value in doing this as well as possible right now, as otherwise Hugo's team might be (rightly) upset that we're not doing a decent enough job. If we're taking custody of this code, and they start using it instead of corelib, we will also become answerable for it!

That said, here is what I am thinking trying to put myself in the API team's shoes. In order to ensure the smoothest integration possible I think that we should make a division based on functionality. So what doesn't make sense to me is a module that has Mongo related code, Solr related code and JAX-ws related code. Because if you just need the mongo classes, you automatically get all these other dependencies that you don't need and could cause weird conflicts that you don't expect. I also don't really like a readme file with possible exclusions, this is not in my opinion something that we can maintain well and just creates overhead.

So let's just create some more modules. Perhaps metis-core can become a parent to several other modules. I'm not saying that we need to do all this as part of this ticket. But I am saying that I don't like the idea of removing the code from corelib until we have had the chance to do this properly (and we can without affecting the classes themselves too much). Just to prevent bothering others with future code changes and dependency annoyances.

What do you think?

So perhaps

Reply:
Ok I've wanted to perform this step by step at a later stage.
But as discussed I will create a more fine grained division of the modules now